packages/components/src/components/hds/text/types.ts

```diff
@@ -0,0 +1,58 @@
+export type HdsTextTags = Extract<keyof HTMLElementTagNameMap, 'span' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'div'>;
+
+export enum HdsTextColorValues {
```

Comment:
Took the approach here of defining arg values with enums first, so that consumers can optionally import these enums directly and avoid the use of magic strings/avoid typos. This would be particularly helpful for consumers who are not yet using glint to type-check templates. Adding the ability for consumers to import these types would happen in a follow up PR.

The enums are then converted to string unions (or string and number unions in the case of size) so that consumers can provide args via strings `@color="primary"` or enum value `@color={{HdsTextColorValues.Primary}}`

Reply:
> do you see any downsides in using the @color={{HdsTextColorValues.Primary}} syntax
I don't think so? Although the enum does need to be made accessible to the template by adding it as a class property

> can you expand on this?
Today, a consumer of HDS components often has to provide string arguments that must be one of a set of values, for example `@color="primary"`. If one were to typo (`@color="prmary"`) or provide the wrong value for an arg (`@color="error"`), they would need to rely on manual QA to catch the mistake. The use of enums doesn't eliminate the possibility of a typo, but could potentially make it more clear what the options for a given arg should be

Reply:
With glint enabled they wouldn't be able to pass in a typo, but I imagine glint won't be adopted across all of cloud-ui for some time. Enums only improve the UX imo.

Reply:
@WenInCode hold my beer 🤣